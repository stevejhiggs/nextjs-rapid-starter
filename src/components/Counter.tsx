import { Box, Button } from '@chakra-ui/react';
import React, { useState } from 'react';

// An example of implementing an interactive component using hooks
const Counter: React.FC = () => {
  const [count, setCount] = useState(0);

  return (
    <Box marginBottom="2rem">
      <p data-testid="message">You clicked {count} times</p>
      <Button data-testid="increment" colorScheme={'blue'} onClick={() => setCount(count + 1)}>
        Increment counter
      </Button>
    </Box>
  );
};

export default Counter;
