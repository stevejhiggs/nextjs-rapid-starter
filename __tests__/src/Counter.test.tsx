import React from 'react';
import { cleanup, render, fireEvent, screen } from '@testing-library/react';
import Counter from '@/src/Counter';

afterEach(cleanup);

describe('Components -> Counter', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Counter />);
    expect(container).toMatchSnapshot();
  });

  it('Should initially have a value of 0', () => {
    render(<Counter />);
    expect(screen.getByTestId('message')).toHaveTextContent('You clicked 0 times');
  });

  it('Should enable the button', () => {
    render(<Counter />);
    expect(screen.getByTestId('increment')).not.toHaveAttribute('disabled');
  });

  it('Should increment the count by 1 when the button is clicked', () => {
    render(<Counter />);
    fireEvent.click(screen.getByTestId('increment'));
    expect(screen.getByTestId('message')).toHaveTextContent('You clicked 1 times');
  });
});
