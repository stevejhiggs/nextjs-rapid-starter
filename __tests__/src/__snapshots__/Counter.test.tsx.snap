// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> Counter should match the snapshot 1`] = `
<div>
  <div
    class="css-8fkoyy"
  >
    <p
      data-testid="message"
    >
      You clicked 
      0
       times
    </p>
    <button
      class="chakra-button css-4xx2wk"
      data-testid="increment"
      type="button"
    >
      Increment counter
    </button>
  </div>
</div>
`;
